apps/dolly-backend/src/main/java/no/nav/dolly/bestilling/kontoregisterservice/KontoregisterConsumer.java

```diff
@@ -0,0 +1,93 @@
+import no.nav.dolly.bestilling.kontoregisterservice.dto.OppdaterKontoRequestDto;
+import no.nav.dolly.bestilling.kontoregisterservice.dto.UtenlandskKontoDto;
+import no.nav.dolly.config.credentials.KontoregisterConsumerProperties;
+import no.nav.dolly.metrics.Timed;
+import no.nav.dolly.security.config.NaisServerProperties;
+import no.nav.testnav.libs.dto.tpsmessagingservice.v1.BankkontonrNorskDTO;
+import no.nav.testnav.libs.dto.tpsmessagingservice.v1.BankkontonrUtlandDTO;
+import no.nav.testnav.libs.dto.tpsmessagingservice.v1.TpsMeldingResponseDTO;
+import no.nav.testnav.libs.standalone.servletsecurity.exchange.TokenExchange;
+import org.springframework.stereotype.Service;
+import org.springframework.web.reactive.function.client.ExchangeFilterFunction;
+import org.springframework.web.reactive.function.client.WebClient;
+
+import java.security.SecureRandom;
+import java.util.List;
+import java.util.Random;
+import java.util.stream.Collectors;
+import java.util.stream.Stream;
+
+import static no.nav.dolly.util.JacksonExchangeStrategyUtil.getJacksonStrategy;
+
+@Slf4j
+@Service
+public class KontoregisterConsumer {
+
+    private static final Random random = new SecureRandom();
+
+    private final WebClient webClient;
+    private final TokenExchange tokenService;
+    private final NaisServerProperties serviceProperties;
+
+    public KontoregisterConsumer(TokenExchange tokenService,
+                                 KontoregisterConsumerProperties serverProperties,
+                                 ObjectMapper objectMapper,
+                                 ExchangeFilterFunction metricsWebClientFilterFunction) {
+
+        this.tokenService = tokenService;
+        this.serviceProperties = serverProperties;
+        this.webClient = WebClient.builder()
+                .baseUrl(serverProperties.getUrl())
+                .exchangeStrategies(getJacksonStrategy(objectMapper))
+                .filter(metricsWebClientFilterFunction)
+                .build();
+    }
+
+    public static String tilfeldigUtlandskBankkonto() {
+        var kontonummerLengde = 15;
+
+        return Stream.concat(
+            random.ints(2, 'A', 'Z')
+                    .boxed()
+                    .map(i -> Character.toString(i.intValue()))
+            ,
+            random.ints(kontonummerLengde, 0, 10)
+                    .boxed()
+                    .map(Integer::toUnsignedString)
+        ).collect(Collectors.joining());
+    }
+
+    @Timed(name = "providers", tags = {"operation", "tps_messaging_createUtenlandskBankkonto"})
+    public List<TpsMeldingResponseDTO> sendUtenlandskBankkontoRequest(String ident, BankkontonrUtlandDTO body) {
+
+        if ((null != body.getTilfeldigKontonummer()) && body.getTilfeldigKontonummer()) {
+            body = body.withKontonummer(tilfeldigUtlandskBankkonto());
+        }
+
+        var utenlandskKontoDto = new UtenlandskKontoDto(
```

Comment:
Er det ikke tydeligere å benytte factory i stedet for constructor? 
Eventuelt benytte Orika og egen klasse for mapping.